core/src/main/java/org/mule/runtime/core/interceptor/AbstractEnvelopeInterceptor.java

```diff
@@ -49,7 +47,7 @@ public abstract MuleEvent last(MuleEvent event, ProcessingTime time, long startT
   @Override
   protected MuleEvent processBlocking(MuleEvent event) throws MuleException {
     long startTime = System.currentTimeMillis();
-    ProcessingTime time = event.getProcessingTime();
+    ProcessingTime time = ((DefaultMessageExecutionContext) event.getExecutionContext()).getProcessingTime();
```

Comment:
We really need to find a cleaner way to avoid this cast (we've been casting it in compatibility module, but shouldn't in core).  If it makes sense to do this later and not now then need to add a TODO with JIRA.